src/contracts/core/AllocationManager.sol

```diff
@@ -55,138 +63,131 @@ contract AllocationManager is
-            require(avsDirectory.isOperatorSetBatch(allocation.operatorSets), InvalidOperatorSet());
-
-            // 1. Check current maxMagnitude matches expected value. This is to check for slashing race conditions
-            // where an operator gets slashed from an operatorSet and as a result all the configured allocations have larger
-            // proprtional magnitudes relative to each other.
-            uint64 maxMagnitude = _maxMagnitudeHistory[msg.sender][allocation.strategy].latest();
-            require(maxMagnitude == allocation.expectedMaxMagnitude, InvalidExpectedMaxMagnitude());
-
-            // 2. For the given (operator,strategy) clear any clearable pending deallocations to free up encumberedMagnitude
-            _clearDeallocationQueue({operator: msg.sender, strategy: allocation.strategy, numToClear: type(uint16).max});
-
-            for (uint256 j = 0; j < allocation.operatorSets.length; ++j) {
-                bytes32 operatorSetKey = _encodeOperatorSet(allocation.operatorSets[j]);
-
-                // Ensure there is not already a pending modification
-                PendingMagnitudeInfo memory info =
-                    _getPendingMagnitudeInfo(msg.sender, allocation.strategy, operatorSetKey);
-                require(info.pendingDiff == 0, ModificationAlreadyPending());
-
-                info.pendingDiff = _calcDelta(info.currentMagnitude, allocation.magnitudes[j]);
-                require(info.pendingDiff != 0, SameMagnitude());
-
-                // Calculate the effectTimestamp for the modification
-                if (info.pendingDiff < 0) {
-                    info.effectTimestamp = uint32(block.timestamp) + DEALLOCATION_DELAY;
-
-                    // Add the operatorSet to the deallocation queue
-                    deallocationQueue[msg.sender][allocation.strategy].pushBack(operatorSetKey);
-                } else if (info.pendingDiff > 0) {
-                    info.effectTimestamp = uint32(block.timestamp) + operatorAllocationDelay;
-
-                    // For allocations, immediately add to encumberedMagnitude to ensure the operator
-                    // can't allocate more than their maximum
-                    info.encumberedMagnitude = _addInt128(info.encumberedMagnitude, info.pendingDiff);
-                    require(info.encumberedMagnitude <= maxMagnitude, InsufficientAllocatableMagnitude());
+        require(delegation.isOperator(msg.sender), InvalidOperator());
+
+        for (uint256 i = 0; i < params.length; i++) {
+            require(params[i].strategies.length == params[i].newMagnitudes.length, InputArrayLengthMismatch());
+
+            // Check that the operator set exists and whether the operator is slashable for the set
+            OperatorSet calldata operatorSet = params[i].operatorSet;
+            require(_operatorSets[operatorSet.avs].contains(operatorSet.id), InvalidOperatorSet());
+            bool isSlashable = isOperatorSlashable(msg.sender, operatorSet);
```

Comment:
Should we also have a IStrategy param passed into `isOperatorSlashable`?
Right now we allow deallocations to immediately decrement encumberedMagnitude if it is not slashable. But what if the AVS removes the specific strategy. We should let the operator deallocate immediately in this scenario too.

Reply:
Okay - should have addressed this here: [62a802a](https://github.com/Layr-Labs/eigenlayer-contracts/pull/860/commits/62a802a305f313d78edd7acce27fecadae611520)

Reply:
> specifically we need to ensure that if the strategy is not a part of the set, your only action is to reclaim an existing allocation

note, i did not implement this portion -- `modifyAllocations` still allows a user to allocate to an operator set even if the set does not support the strategy. Was simpler to implement this way and should be instantly deallocatable after the allocation delay anyway, so nbd.